### ISSUE TITLE
Backend: Nil pointer deref and logout button does not work

### ISSUE THREAD

**Issue by ishanjain28** [grafana/grafana#87103]
### What happened?

When I try to logout using Profile Image -> Logout, Grafana sends a 502 error and refreshing the page shows I was not logged out at all.

<details>
<summary>In the application logs, I see this error & stacktrace.</summary>

```
Apr 30 10:24:18 grafana grafana-server[176]: logger=context userId=1 orgId=1 uname=ishan t=2024-04-30T10:24:18.266061343+05:30 level=error msg="Request error" error="runtime error: invalid memory address or nil pointer dereference" stack="runtime/panic.go:261 (0x5035ef7)\nruntime/signal_unix.go:861 (0x5035ec5)\ngithub.com/grafana/grafana/pkg/api/login.go:263 (0x87af8a2)\ngithub.com/grafana/grafana/pkg/api/response/web_hack.go:40 (0x615608f)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:131 (0x6148b2e)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/middleware/csp.go:50 (0x672b0ce)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/middleware/csp.go:41 (0x672ae06)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/services/contexthandler/contexthandler.go:145 (0x87cf79b)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/render.go:44 (0x61496bd)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/macaron.go:137 (0x6148bd2)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/middleware/csrf/csrf.go:66 (0x7e1eb7a)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/middleware/recovery.go:179 (0x87a9f75)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/middleware/gziper.go:76 (0x87ceb45)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/middleware/loggermw/logger.go:72 (0x7e1fb96)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/middleware/request_metrics.go:83 (0x672d735)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/middleware/request_tracing.go:88 (0x87aabd5)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/middleware/requestmeta/request_metadata.go:66 (0x87ce835)\nnet/http/server.go:2136 (0x538ee28)\ngithub.com/grafana/grafana/pkg/web/context.go:52 (0x6147616)\ngithub.com/grafana/grafana/pkg/web/router.go:155 (0x614ad6d)\ngithub.com/grafana/grafana/pkg/web/router.go:211 (0x614b8d4)\ngithub.com/grafana/grafana/pkg/web/macaron.go:163 (0x6148ffc)\nnet/http/server.go:2938 (0x5391b0d)\nnet/http/server.go:2009 (0x538d9f3)\nruntime/asm_amd64.s:1650 (0x50553e0)\n"
Apr 30 10:24:18 grafana grafana-server[176]: server.go:3214: http: panic serving 10.0.50.3:34452: Context.HTML - Error rendering template: error. You may need to build frontend assets
Apr 30 10:24:18 grafana grafana-server[176]:  template: error:16:35: executing "error" at <.ContentDeliveryURL>: can't evaluate field ContentDeliveryURL in type struct { Title string; AppTitle string; AppSubUrl string; Theme string; ErrorMsg string; Assets *dtos.EntryPointAssets }
Apr 30 10:24:18 grafana grafana-server[176]: goroutine 146900 [running]:
Apr 30 10:24:18 grafana grafana-server[176]: net/http.(*conn).serve.func1()
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:1868 +0xb9
Apr 30 10:24:18 grafana grafana-server[176]: panic({0x597c20?, 0xc00419f830?})
Apr 30 10:24:18 grafana grafana-server[176]:         runtime/panic.go:920 +0x270
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.(*Context).HTML(0xc0020a9940, 0x13c3cf8?, {0x13a1d6e, 0x5}, {0xfb2820, 0xc004f6af00})
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/context.go:112 +0x248
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/api.(*HTTPServer).addMiddlewaresAndStaticRoutes.Recovery.func4.1.1()
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/middleware/recovery.go:174 +0x668
Apr 30 10:24:18 grafana grafana-server[176]: panic({0x94c1c0?, 0x97e5f20?})
Apr 30 10:24:18 grafana grafana-server[176]:         runtime/panic.go:914 +0x21f
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/api.(*HTTPServer).Logout(0xc0029cc000, 0xc002658380)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/api/login.go:263 +0x2c2
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/api/response.wrap_handler.func3({0x1fc61c0?, 0xc00174acd0?}, 0x1fde988?)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/api/response/web_hack.go:40 +0x30
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0xc003a9e4b0?, {0x1fc61c0?, 0xc00174acd0?}, 0x510cf85?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f9500)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:131 +0x12f
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x5159200?, {0x1fc61c0?, 0xc00174acd0?}, 0xc00268c360?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0xde0d00?, {0x1fc61c0?, 0xc00174acd0?}, 0x17?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/middleware.cspMiddleware.func1({0x1fc61c0, 0xc00174acd0}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/middleware/csp.go:50 +0x1af
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0xc0027468a0?, {0x1fc61c0?, 0xc00174acd0?}, 0xc0020d52a0?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/middleware.nonceMiddleware.func1({0x1fc61c0, 0xc00174acd0}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/middleware/csp.go:41 +0x247
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x951f20?, {0x1fc61c0?, 0xc00174acd0?}, 0x510cf85?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x4ffbcc8?, {0x1fc61c0?, 0xc00174acd0?}, 0x510cf85?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x18?, {0x1fc61c0?, 0xc00174acd0?}, 0xc00268c2d0?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0xc002d12b40?, {0x1fc61c0?, 0xc00174acd0?}, 0x6?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/services/contexthandler.(*ContextHandler).Middleware-fm.(*ContextHandler).Middleware.func1({0x1fc61c0, 0xc00174acd0}, 0xc002a9a700?)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/services/contexthandler/contexthandler.go:145 +0xbdc
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x1fc61c0?, {0x1fc61c0?, 0xc00174acd0?}, 0x510cf85?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x1fc61c0?, {0x1fc61c0?, 0xc00174acd0?}, 0x510cf85?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x1fc61c0?, {0x1fc61c0?, 0xc00174acd0?}, 0x510cf85?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x1fc61c0?, {0x1fc61c0?, 0xc00174acd0?}, 0x510cf85?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x9870e60?, {0x1fc61c0?, 0xc00174acd0?}, 0x510cf85?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x4fe9f46?, {0x1fc61c0?, 0xc00174acd0?}, 0x20?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.Renderer.func1.1({0x1fc61c0, 0xc00174acd0}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/render.go:44 +0x21e
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x1f98e60?, {0x1fc61c0?, 0xc00174acd0?}, 0x510cf85?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x1f98e60?, {0x1fc61c0?, 0xc00174acd0?}, 0x510cf85?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x1f98e60?, {0x1fc61c0?, 0xc00174acd0?}, 0x510cf85?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc61c0?, 0xc00174acd0?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x530a94d?, {0x1fc61c0?, 0xc00174acd0?}, 0x13a810e?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.mwFromHandler.func1.1({0x1fc4c30?, 0xc00267b098?}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:137 +0x1d3
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0xc0029c2de0?, {0x1fc4c30?, 0xc00267b098?}, 0x4ffbcc8?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/middleware/csrf.(*CSRF).Middleware.func1.1({0x1fc4c30, 0xc00267b098}, 0x98b4dc0?)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/middleware/csrf/csrf.go:66 +0x1fb
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x1fde988?, {0x1fc4c30?, 0xc00267b098?}, 0xc002a9af50?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/api.(*HTTPServer).addMiddlewaresAndStaticRoutes.Recovery.func4.1({0x1fc4c30, 0xc00267b098}, 0xc002978c90?)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/middleware/recovery.go:179 +0xb6
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x754ec4bc86d8?, {0x1fc4c30?, 0xc00267b098?}, 0x13cdef4?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/api.(*HTTPServer).addMiddlewaresAndStaticRoutes.Gziper.func3.1({0x1fc61c0?, 0xc00174abe0}, 0xc0025f8c00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/middleware/gziper.go:76 +0x226
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x1fde988?, {0x1fc61c0?, 0xc00174abe0?}, 0x13d95b1?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/middleware/loggermw.(*loggerImpl).Middleware.func1.1({0x1fc61c0?, 0xc00174abe0}, 0xc0025f8a00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/middleware/loggermw/logger.go:72 +0x5b7
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x0?, {0x1fc61c0?, 0xc00174abe0?}, 0x0?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/middleware.RequestMetrics.func1.1({0x1fc61c0?, 0xc00174abe0}, 0xc0025f8a00)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/middleware/request_metrics.go:83 +0x256
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0xc0025f8800?, {0x1fc61c0?, 0xc00174abe0?}, 0xc001433bf0?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/api.(*HTTPServer).addMiddlewaresAndStaticRoutes.RequestTracing.func2.1({0x1fc61c0, 0xc00174abe0}, 0xc0025f8800)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/middleware/request_tracing.go:88 +0x416
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0xc0025f8800?, {0x1fc61c0?, 0xc00174abe0?}, 0x7?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/api.(*HTTPServer).addMiddlewaresAndStaticRoutes.SetupRequestMetadata.func1.1({0x1fc61c0, 0xc00174abe0}, 0xc0025f8800)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/middleware/requestmeta/request_metadata.go:66 +0x176
Apr 30 10:24:18 grafana grafana-server[176]: net/http.HandlerFunc.ServeHTTP(0x1f94400?, {0x1fc61c0?, 0xc00174abe0?}, 0xc003791920?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2136 +0x29
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.(*Context).run(0xc0020a9940)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/context.go:52 +0xf7
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.(*Router).Handle.func1({0x1fc4d80, 0xc002720d20}, 0x139f0e6?, 0xc00274679c?)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/router.go:155 +0x14e
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.(*Router).ServeHTTP(0xc001a2b3c0, {0x1fc4d80, 0xc002720d20}, 0xc0025f8600)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/router.go:211 +0x1b5
Apr 30 10:24:18 grafana grafana-server[176]: github.com/grafana/grafana/pkg/web.(*Macaron).ServeHTTP(0xc0029c39b0?, {0x1fc4d80?, 0xc002720d20?}, 0xc0025f8600?)
Apr 30 10:24:18 grafana grafana-server[176]:         github.com/grafana/grafana/pkg/web/macaron.go:163 +0xdd
Apr 30 10:24:18 grafana grafana-server[176]: net/http.serverHandler.ServeHTTP({0xc002978240?}, {0x1fc4d80?, 0xc002720d20?}, 0x6?)
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2938 +0x8e
Apr 30 10:24:18 grafana grafana-server[176]: net/http.(*conn).serve(0xc00287fcb0, {0x1fde988, 0xc002d339b0})
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:2009 +0x5f4
Apr 30 10:24:18 grafana grafana-server[176]: created by net/http.(*Server).Serve in goroutine 468
Apr 30 10:24:18 grafana grafana-server[176]:         net/http/server.go:3086 +0x5cb
Apr 30 10:24:18 grafana grafana-server[176]: logger=context userId=1 orgId=1 uname=ishan t=2024-04-30T10:24:18.389559894+05:30 level=info msg="Request Completed" method=GET path=/favicon.ico status=404 remote_addr=2a0a:6040:4004:10:8ad9:4281:7425:ea94 time_ms=4 duration=4.898223ms size=46425 referer=https://dash.ishanjain.me/logout handler=public-assets
Apr 30 10:24:19 grafana grafana-server[176]: logger=context userId=1 orgId=1 uname=ishan t=2024-04-30T10:24:19.85546535+05:30 level=info msg="Request Completed" method=GET path=/api/live/ws status=-1 remote_addr=2a0a:6040:4004:10:8ad9:4281:7425:ea94 time_ms=0 duration=811.248µs size=0 referer= handler=/api/live/ws
```

</details>

My config is attached in this thread. This might be happening because of a misconfiguration but I can't spot it and I am not sure if it's misconfig or some thing else is broken.

### What did you expect to happen?

I should be able to logout successfully and I can't unless I manually clear cookies.

### Did this work before?

No, It is broken for as long as I can remember.

### How do we reproduce it?

<details>
<summary>Config</summary>

```
app_mode = production
[paths]
data = /var/lib/grafana
logs = /var/log/grafana
[server]
http_port = 80
domain = dash.ishanjain.me
enforce_domain = true
root_url = https://dash.ishanjain.me
enable_gzip = true
[database]
type = postgres
host = 10.0.50.7:5432
name = grafana
user = grafana
password = 
[datasources]
[remote_cache]
[dataproxy]
[analytics]
[security]
disable_initial_admin_creation = false
secret_key = 
disable_gravatar = true
cookie_secure = true
cookie_samesite = disabled
allow_embedding = false
strict_transport_security = true
strict_transport_security_max_age_seconds = 86400
strict_transport_security_preload = false
strict_transport_security_subdomains = false
x_xss_protection = true
content_security_policy = true
[security.encryption]
[snapshots]
[dashboards]
versions_to_keep = 200
[users]
allow_sign_up = false
[auth]
login_cookie_name = igrafana_session
[auth.anonymous]
hide_version = true
[auth.github]
[auth.gitlab]
[auth.google]
[auth.grafana_com]
[auth.azuread]
[auth.okta]
[auth.generic_oauth]
[auth.basic]
[auth.proxy]
[auth.jwt]
[auth.ldap]
enabled = false
config_file = /etc/grafana/ldap.toml
[aws]
[azure]
[rbac]
[smtp]
[emails]
[log]
[log.console]
[log.file]
[log.syslog]
[log.frontend]
[quota]
[unified_alerting]
[unified_alerting.reserved_labels]
[alerting]
[annotations]
[annotations.dashboard]
[annotations.api]
[explore]
[help]
enabled = false
[profile]
[query_history]
[metrics]
[metrics.environment_info]
[metrics.graphite]
[grafana_com]
[tracing.jaeger]
[tracing.opentelemetry]
[tracing.opentelemetry.jaeger]
[tracing.opentelemetry.otlp]
[external_image_storage]
[external_image_storage.s3]
[external_image_storage.webdav]
[external_image_storage.gcs]
[external_image_storage.azure_blob]
[external_image_storage.local]
[rendering]
[panels]
[plugins]
[live]
[plugin.grafana-image-renderer]
[enterprise]
[feature_toggles]
[date_formats]
[expressions]
[geomap]
[navigation.app_sections]
[navigation.app_standalone_pages]
```
</details>

1. Use the above config to start grafana. I use ldap but in the config, I have disabled ldap and enabled initial account creation.
2. Login to grafana and then use Profile -> Logout to logout of it
3. It returns a 502 error instead of logging you out and redirecting to login page.

### Is the bug inside a dashboard panel?

No

### Environment (with versions)?

Grafana: v10.2.3 (1e84fede543acc892d2a2515187e545eb047f237)
OS: Arch linux(`Linux emerald 6.8.7-arch1-2 #1 SMP PREEMPT_DYNAMIC Fri, 19 Apr 2024 09:51:31 +0000 x86_64 GNU/Linux
`) . Also tested on Android phones and ipads. Same problem on all of them
Browser: Firefox 125.0.2 / Chromium 126.0.6444.0

### Grafana platform?

A package manager (APT, YUM, BREW, etc.)

**Comment by tonypowa** [comment]
Hi @ishanjain28 

Thank you for the detailed report and for providing configuration details 

Moving issue to the squad's board for review

**Comment by mgyongyosi** [comment]
@ishanjain28 This is a known bug in 10.2.3 (https://github.com/grafana/grafana/issues/79839#issuecomment-1883098888). 

It's been fixed in newer versions of Grafana.

**Comment by ishanjain28** [comment]
Okay, I'll update and test this again!

**Comment by ishanjain28** [comment]
This has been fixed as of Grafana v11.0.0. Closing
